src/transformers/models/whisper/tokenization_whisper_fast.py

```diff
@@ -320,12 +321,17 @@ def decode(
             return {"text": text, "offsets": offsets}
         return text
 
-    def _decode(self, *args, normalize: bool = False, **kwargs) -> str:
+    def _decode(self, *args, normalize: bool = False, decode_with_timestamps: bool = False, **kwargs) -> str:
         if kwargs["skip_special_tokens"]:
             prompt_token_id = self.convert_tokens_to_ids("<|startofprev|>")
             decoder_start_token_id = self.convert_tokens_to_ids("<|startoftranscript|>")
             kwargs["token_ids"] = self._strip_prompt(kwargs["token_ids"], prompt_token_id, decoder_start_token_id)
 
+        if not decode_with_timestamps:
+            # filter timestamp tokens if they are contained in the vocab
+            timestamp_ids = self.convert_tokens_to_ids([("<|%.2f|>" % (i * 0.02)) for i in range(1500 + 1)])
+            kwargs["token_ids"] = [token for token in kwargs["token_ids"] if token not in timestamp_ids]
```

Comment:
I guess it's this `if in` logic that is causing the fast tokenizer tests to take > 120s? If so, I'll move this logic out of `_decode` and into `decode`

Reply:
yes, in `_decode` we should not evven have the `_strip_prompt` and etc, since the special tokens are/should be processed in the parent `decode`

Reply:
Awesome - thanks for the clarification!